pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -160,9 +160,13 @@ func (c *Cluster) Start() error {
 				return errors.Wrapf(err, "failed to create mgr deployment %s", resourceName)
 			}
 			logger.Infof("deployment for mgr %s already exists. updating if needed", resourceName)
-
-			if err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, false); err != nil {
-				logger.Errorf("failed to update mgr deployment %q. %v", resourceName, err)
+			if err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy); err != nil {
+				if c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy {
+					logger.Errorf("failed to update mgr deployment %q. %v", resourceName, err)
```

Comment:
```suggestion
					logger.Errorf("failed to update mgr deployment %q. %v", resourceName, err)
```
this can moved to before L164 i.e before `if c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy {`

Reply:
@subhamkrai  wrote:

> this can moved to before L164 i.e before `if c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy {`


sure, it can be moved. but why? I. E. what's the point? 
Not meaning to sound rude here, just want to understand  the reasoning ... 
I see that the error wrapping in the else branch is very similar, but it does not create the identical log message

Reply:
maybe remove the `else` statement as it is only logging error

Reply:
@subhamkrai  wrote:
> The reason, IMO if there is a error we always want to log error messages and only when `if c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy` is `true`.

good point! I'll move it, but I am also confused because in my local commit, I have added an additional Info log message that we are continuing the reconcile. Not sure why again, github is not showing that hunk ...